spec/lib/reports/careplus_exporter_spec.rb

```diff
@@ -10,25 +10,49 @@
     )
   end
 
+  let(:programme) { create(:programme, programme_type) }
+  let(:programme_type) { nil }
+  let(:parsed_csv) { CSV.parse(csv) }
+  let(:programmes) { [programme] }
+  let(:session) { create(:session, organisation:, programmes:, location:) }
+  let(:location) do
+    create(
+      :school,
+      gias_local_authority_code: 123,
+      gias_establishment_number: 456
+    )
+  end
+  let(:headers) { parsed_csv.first }
+  let(:organisation) do
+    create(:organisation, careplus_venue_code: "ABC", programmes:)
+  end
+  let(:patient_session) do
+    create(
+      :patient_session,
+      :consent_given_triage_not_needed,
+      programmes:,
+      session:
+    )
+  end
+  let(:vaccine) { create(:vaccine, programme:) }
+
+  def create_vaccination_record(attributes = {})
+    create(
+      :vaccination_record,
+      {
+        programme:,
+        vaccine:,
+        patient: patient_session.patient,
+        session: patient_session.session,
+        performed_at: 2.weeks.ago
+      }.merge(attributes)
+    )
+  end
```

Comment:
I think instead of having a method like this, there could be an option to refactor the `shared_examples`, so they can be used like this:

```rb
it_behaves_like "generates a report", "flu" do
  let(:programme) { create(:programme, :flu) }
  let(:delivery_method) { :intramuscular }
end

it_behaves_like "generates a report", "flu nasal spray" do
  let(:programme) { create(:programme, :flu) }
  let(:delivery_method) { :nasal_spray }
end
```

I think this would be a more conventional way of structuring these tests.

Reply:
I agree 👍